### ISSUE TITLE
Subresource-integrity loading behaviour with malformed integrity hashes

### ISSUE THREAD

**Issue by vogtm** [web-platform-tests/interop#666]
While this might be quite a niche topic, it seems like an interop issue. I hope this is the right place to document it 

_**Considering this simple setup:**_ 
```html
<head>
  <link rel="stylesheet" href="demo.css" >
</head>
```
It's loading normally.

---

```html
<head>
  <link rel="stylesheet" href="demo.css" integrity="..." crossorigin="anonymous">
</head>
```
The ressource will be allowed in Chrome, FF and Safari.

---

```html
<head>
   <link rel="stylesheet" href="demo.css" integrity="sha256-AAA" crossorigin="anonymous">
</head>
```
The ressource will be blocked in all thre browers.

---

```html
<head>
  <link rel="stylesheet" href="demo.css" integrity="sha256-..." crossorigin="anonymous">
</head>
```
The ressource will be allowed in Chrome and Safari but blocked in FF.

---

I wasn't really able to clearly pinpoint what should be the correct behaviour. 
I think this is not completely clear and still in open discussion https://github.com/w3c/webappsec-subresource-integrity/issues/120
But still a clear difference between browsers.

**Comment by foolip** [comment]
@vogtm sorry your issue was left just sitting for almost a year, I'm doing some issue triage now and came across this.

I think the spec for this is https://w3c.github.io/webappsec-subresource-integrity/#the-integrity-attribute and that the problem with the `integrity="sha256-..."` must be that Firefox treats "..." as an invalid or mismatching SHA-256 hash, instead of rejecting the whole "sha256-..." string as not matching the ABNF grammar in the spec.

In other words, I think Firefox isn't following the spec here. I couldn't find any tests for this, so I sent https://github.com/web-platform-tests/wpt/pull/50875 using your example.

**Comment by vogtm** [comment]
@foolip no worries, not something thats urgent and probably not something a lot of people will effect at all. 

Thank you very much for taking care of it.

**Comment by foolip** [comment]
The tests for this have landed in https://github.com/web-platform-tests/wpt/pull/50875 now. I'll go ahead and close this. Thanks @vogtm for reporting!
